File: sdk/python/alcedo_sdk/resources/activity_logs.py
from typing import Any, Optional

from .base import BaseResource


class ActivityLogsResource(BaseResource):
    async def list_collections(self, params: Optional[dict[str, str]] = None) -> dict:
        return await self._request("GET", "/api/activity-logs/collections", params=params)

    async def list_system(self, params: Optional[dict[str, str]] = None) -> dict:
        return await self._request("GET", "/api/activity-logs/system", params=params)

File: sdk/alcedo-sdk/src/activityLogs.ts
export function createActivityLogsResource(ky: any) {
  return {
    listCollections: (params?: URLSearchParams, options?: any) =>
      ky.get("logs/collections", { searchParams: params as any, ...options }).json(),
    listSystem: (params?: URLSearchParams, options?: any) =>
      ky.get("logs/system", { searchParams: params as any, ...options }).json(),
  };
}
